File: problems/2971_find-polygon-with-the-largest-perimeter/__main__.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        ret, pre_sum = -1, 0
        for num in nums:
            if pre_sum > num:
                ret = pre_sum + num
            pre_sum += num
        return ret
        

File: problems/2971_find-polygon-with-the-largest-perimeter/main.go
func largestPerimeter(nums []int) int64 {
    sort.Ints(nums)
    ret, pre_sum := int64(-1), int64(0)
    for _, num := range nums {
        tmp := int64(num)
        if pre_sum > tmp {
            ret = pre_sum + tmp
        }
        pre_sum += tmp
    }
    return ret
}